src/30_Advanced/Geolocation_with_amp-list.html

```diff
@@ -0,0 +1,140 @@
+    .abe-preview p.title {
+      font-weight: bold;
+      line-height: 16px;
+      margin: 10px;
+    }
+
+    .abe-preview ul.results {
+      list-style: none;
+      margin: 0;
+      padding: 0 10px;
+      overflow: auto; /* Fixes collapsing margins on the LIs */
+    }
+
+    .abe-preview ul.results li {
+      height: 100px;
+      background: #fff;
+      border-radius: 2px;
+      margin-bottom: 15px;
+      position: relative;
+      box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
+    }
+    .abe-preview ul.results p {
+      margin: 0;
+    }
+    .abe-preview ul.results amp-img {
+      background-image: url(/img/geolocation_map_skeleton.png);
+      background-size: 150px 100px;
+      float: left;
+      margin-right: 5px;
+    }
+    .abe-preview div[placeholder] ul.results li {
+      background: url(/img/geolocation_map_skeleton.png), /* Image */
+        linear-gradient(#cccccc 100%, transparent 0), /* Image Placeholder */
+        linear-gradient(#bbbbbb 100%, transparent 0), /* City */
+        linear-gradient(#cccccc 100%, transparent 0), /* State */
+        linear-gradient(#cccccc 100%, transparent 0); /* Country */
+      background-size: 150px 100px, /* Image */
+        150px 100px, /* Image Placeholder */
+        150px 15px, /* City */
+        130px 15px, /* State */
+        130px 15px; /* Country */
+      background-position: 0 0, /* Image */
+        0 0, /* Image Placholder */
+        155px 4px, /* City */
+        155px 29px, /* State */
+        155px 53px; /* Country */
+      background-repeat: no-repeat;
+    }
+    .abe-preview span.placeholder {
+      display: inline-block;
+      height: 14px;
+      width: 150px;
+      background-color: #cccccc;
+      position: relative;
+      top: 3px;
+    }
+  </style>
+</head>
+<body>
+
+  <!-- ## Geo-based Search Results -->
+  <!-- `amp-list` makes a request to a server backend which does IP-based
+  geolocation and returns relevant search results. For this demo, it calls an
+  API to get nearby cities. The backend returns JSON results which `amp-list`
+  renders with `amp-mustache`. The backend makes two remote API calls (and also
+  adds 500ms of simulated delay); to mitigate the UX effects of this delay we
+  use `amp-list`'s `placeholder` to lay out a skeleton.
+
+  Since we need to have an initial fixed-size for amp-list we manually calculate
+  that each card is 100 + 15px margin and the title is 16px + 20px margin:
+
+  > `16px + 20px + (5 * (100 + 15px)) = 611px`
+
+  The placeholder list items largely use CSS to draw the skeleton elements as
+  described in [CSS-Tricks](https://css-tricks.com/building-skeleton-screens-css-custom-properties/).
+  The skeleton includes a tiny blurred map to add to the realism, and that map
+  is also used while `<amp-img>` loads the actual images to prevent a
+  flash when the placeholder is removed.
+  -->
+  <amp-list class="abe-preview" width="auto" height="611" layout="fixed-height"
```

Comment:
Suggestion: add the 'noloading' attribute to hide the default loading indicator.

Reply:
Cool! I wasn't aware of that. (I think it should be automatic if you're using placeholder.) Done.